### PR TITLE
HW2 PR1 Dockerfile and docker registry

### PR DESCRIPTION
**Pull request by yevhen-k**
Hi @truskovskiyk !
Here is a simple docker file and a link to the docker registry where built image is pushed to.

https://hub.docker.com/r/yevhenk10s/prjctr02-infrastructure

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Flask==2.1.3
+# torch==1.12.0
+# huggingface-hub==0.8.1
+# transformers==4.20.1
+# gdown==4.5.1
```
Comment (truskovskiyk) [inline requirements.txt]: remove commented code

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM python:3.10-alpine3.16
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+# RUN apt-get update -y && apt-get upgrade -y
```
Comment (truskovskiyk) [inline Dockerfile]: remove commented code

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+WORKDIR /app
+COPY requirements.txt requirements.txt
+RUN pip3 install --upgrade pip && pip3 install -r requirements.txt
+COPY . .
+EXPOSE 5000
+CMD ["python3", "app.py"]
```
Comment (truskovskiyk) [inline Dockerfile]: optional: would be nice if you switch user after installing all deps

Comment (truskovskiyk) [inline Dockerfile]: consider using [gunicorn](https://github.com/benoitc/gunicorn) for production deployment

File: app.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+@app.route('/', methods=['GET', 'POST'])
+def process():
+    if request.method == 'GET':
+        return Response(response=json.dumps({"response": "ok"}), status=200, mimetype="application/json")
+    else:
+        texts = request.json["texts"]
+        response = model.process(texts)
+        return Response(response=json.dumps({"response": response}), status=200, mimetype="application/json")
+
```
Comment (truskovskiyk) [inline app.py]: `jsonify` should do the same: https://flask.palletsprojects.com/en/2.2.x/api/#flask.json.jsonify

**Review by truskovskiyk — APPROVED** [review]
@yevhen-k great job! please fix comments

File: Dockerfile (hunk 3 above)
Comment (yevhen-k) [reply Dockerfile]: Yeah, for the purpose of simplicity and making HW done as quickly as possible, there were no gunicorn, uwsgi, nginx, etc implemented. But in real world those technologies are must have.

File: requirements.txt (hunk 1 above)
Comment (yevhen-k) [reply requirements.txt]: Resolved with commit e01653cd51757d452ac3c6c6815815c4e8717415

File: Dockerfile (hunk 2 above)
Comment (yevhen-k) [reply Dockerfile]: Resolved with commit e01653cd51757d452ac3c6c6815815c4e8717415

File: Dockerfile (hunk 3 above)
Comment (yevhen-k) [reply Dockerfile]: Resolved with commit e01653cd51757d452ac3c6c6815815c4e8717415

**Pull request merged**